### PR TITLE
fix: adding missing method for searching agencies with topic id for t…

### PR DESCRIPTION
**Pull request by tkuzynow**
…he tenant aware case

Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  @Test
+  void findByPostCodeAndConsultingTypeId_Should_findAgencyByPostcodeAndConsultingType() {
+    // given, when
+    var agencyList = agencyTenantAwareRepository.findByPostCodeAndConsultingTypeId("45501", 5, 20, 1L);
+    // then
+    assertThat(agencyList).hasSize(1);
+  }
+
+  @Test
+  void findByPostCodeAndConsultingTypeId_Should_NotFindAgencyByPostcodeAndConsultingTypeIfTenantIdDoesNotMatch() {
```
Comment (patric-dosch-vi) [inline src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: didn't we agreed on "[...]_When_tenantIdDoesNotMatch" ???

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+class AgencyRepositoryTenantAwareIT {
+
+  @Autowired private AgencyTenantAwareRepository agencyTenantAwareRepository;
+
+  @Test
+  void findByPostCodeAndConsultingTypeId_Should_findAgencyByPostcodeAndConsultingType() {
+    // given, when
+    var agencyList = agencyTenantAwareRepository.findByPostCodeAndConsultingTypeId("45501", 5, 20, 1L);
```
Comment (patric-dosch-vi) [inline src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: general: I have actually always found it helpful in tests when the values were assigned to a well speaking variable. At least the important ones. so e.g. tenantId, notMatchingTenantId, etc. 

No idea if that would help you.

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+@TestPropertySource(properties = "multitenancy.enabled=true")
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.ANY)
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+class AgencyRepositoryTenantAwareIT {
+
+  @Autowired private AgencyTenantAwareRepository agencyTenantAwareRepository;
+
+  @Test
+  void findByPostCodeAndConsultingTypeId_Should_findAgencyByPostcodeAndConsultingType() {
```
Comment (patric-dosch-vi) [inline src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: we find Agencies not only one Agency, I would name them findAgenciesBy...

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java (hunk 2 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: I understand the point, I think for some tests it's helpful, but for these particular tests, as the methods are kept small and simple, I do not see the added value for it

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java (hunk 1 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: fixed

File: src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java (hunk 3 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyRepositoryTenantAwareIT.java]: fixed

**Review by patric-dosch-vi — APPROVED** [review]
hey @tkuzynow, approved.

**Pull request merged**